var/spack/repos/builtin/packages/mptensor/package.py

```diff
@@ -68,33 +68,41 @@ def cmake_args(self):
     def setup_build_tests(self):
         """Copy the build test files after the package is installed to an
         install test subdirectory for use during `spack test run`."""
+
+        # Tests only supported when spec built with mpi
+        if "+mpi" not in self.spec:
+            print("Skipping copy of stand-alone test files: requires +mpi build")
+            return
+
         self.cache_extra_test_sources(".")
 
-    def test(self):
+        # Clean cached makefiles now so only done once
+        print("Converting cached Makefile for stand-alone test use")
+        with working_dir(join_path(install_test_root(self), "tests"), create=False):
+            make("clean")
+            makefile = FileFilter("Makefile")
+            makefile.filter("g++", f"{spack_cxx}", string=True)
+
+        print("Converting cached Makefile.option for stand-alone test use")
+        with working_dir(join_path(install_test_root(self)), create=False):
+            makefile = FileFilter("Makefile.option")
+            makefile.filter("CXX =.*", f"CXX ={self.spec['mpi'].mpicxx}")
+            makefile.filter("CXXFLAGS =.*", f"CXXFLAGS ={self.compiler.cxx11_flag}")
+
+    def test_mptensor(self):
```

Comment:
Adding the name of the package to `test_` is not descriptive of what is actually being tested.